File: BaekJoon Online Judge/C++/Step09-Math_2 Prime and Geometric/1978.cpp
//
//  1978.cpp
//  Step09-Math_2 Prime and Geometric
//
//  Created by Yebin Kim on 2019/10/24.
//  Copyright © 2019 김예빈. All rights reserved.
//

#include <iostream>

using namespace std;

bool searchPrime(int);

// Time Complexity O(n*log(n))
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, check, cnt = 0;
    
    cin >> n;
    
    while(n--) {
        cin >> check;
        
        if(searchPrime(check))
            cnt++;
    }
    
    cout << cnt << "\n";
    
    return 0;
}

bool searchPrime(int input) {
    if(input < 2) return false;
    
    for(int i = 2; i * i <= input; i++)
        if(input % i == 0)
            return false;
    
    return true;
}

File: BaekJoon Online Judge/Swift/Step09-Math_2 Prime and Geometric/1978.swift
//
//  1978.swift
//  Step09-Math_2 Prime and Geometric
//
//  Created by Yebin Kim on 2019/11/17.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

func findPrime(_ num: Int) -> Bool {
    if num < 2 {
        return false
    }
    
    var i = 2
    
    while i * i <= num {
        if num % i == 0 {
            return false
        }
        
        i += 1
    }
    
    return true
}

if let _ = Int(readLine()!) {
    let input = readLine()!.split(separator: " ").map{ Int($0)! }.filter{ findPrime($0) == true }
    print(input.count)
}
